app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt

```diff
@@ -200,6 +219,43 @@ class SplashActivityPresenter @Inject constructor(
   }
 
   private fun processStartupMode() {
+    if (enableAppAndOsDeprecation.value) {
```

Comment:
Wrspping the whole block dependent on the  feature flag value would likely mean that this block is not executed unless the feature flag is enabled/true.

an option is to switch to a `when` block without a varable, or do a seperate `when` block for when the `enableAppAndOsDeprecation` is true

Reply:
I have a seperate block for the legacy deprecation feature. I have also moved the logic for the new deprecation feature to a seperate function so that it's clearer now.